controllers/barbicanapi_controller.go

```diff
@@ -270,129 +267,54 @@ func (r *BarbicanAPIReconciler) generateServiceConfigs(
 	}
 
-	// Fetch the service config snippet (CustomConfigFileName) from the top
-	// level barbican controller, and add them to this service specific Secret.
+	// Fetch the two service config snippets (DefaultsConfigFileName and
+	// CustomConfigFileName) from the Secret generated by the top level
+	// barbican controller, and add them to this service specific Secret.
 	owner := barbican.GetOwningBarbicanName(instance)
 	if owner != "" {
 		barbicanSecretName := owner + "-config-data"
 		barbicanSecret, _, err := secret.GetSecret(ctx, h, barbicanSecretName, instance.Namespace)
 		if err != nil {
 			return err
 		}
+		customData[barbican.DefaultsConfigFileName] = string(barbicanSecret.Data[barbican.DefaultsConfigFileName])
 		customData[barbican.CustomConfigFileName] = string(barbicanSecret.Data[barbican.CustomConfigFileName])
+
+		// TODO(alee) Get custom config overwrites from the parent barbican
 	}
 
 	for key, data := range instance.Spec.DefaultConfigOverwrite {
 		customData[key] = data
 	}
 
-	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
-	// KeystoneAPI not available we should not aggregate the error and continue
-	if err != nil {
-		return err
-	}
-	keystoneInternalURL, err := keystoneAPI.GetEndpoint(endpoint.EndpointInternal)
-	if err != nil {
-		return err
-	}
-
-	ospSecret, _, err := secret.GetSecret(ctx, h, instance.Spec.Secret, instance.Namespace)
-	if err != nil {
-		return err
-	}
-
-	transportURLSecret, _, err := secret.GetSecret(ctx, h, instance.Spec.TransportURLSecret, instance.Namespace)
-	if err != nil {
-		return err
+	customSecrets := ""
+	for _, secretName := range instance.Spec.CustomServiceConfigSecrets {
+		secret, _, err := secret.GetSecret(ctx, h, secretName, instance.Namespace)
+		if err != nil {
+			return err
+		}
+		for _, data := range secret.Data {
+			customSecrets += string(data) + "\n"
+		}
 	}
+	customData[barbican.CustomServiceConfigSecretsFileName] = customSecrets
 
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 
-	databaseAccount := db.GetAccount()
-	databaseSecret := db.GetSecret()
-
 	templateParameters := map[string]interface{}{
-		"DatabaseConnection": fmt.Sprintf("mysql+pymysql://%s:%s@%s/%s?read_default_file=/etc/my.cnf",
-			databaseAccount.Spec.UserName,
-			string(databaseSecret.Data[mariadbv1.DatabasePasswordSelector]),
-			instance.Spec.DatabaseHostname,
-			barbican.DatabaseName,
-		),
-		"KeystoneAuthURL":  keystoneInternalURL,
-		"ServicePassword":  string(ospSecret.Data[instance.Spec.PasswordSelectors.Service]),
-		"ServiceUser":      instance.Spec.ServiceUser,
-		"ServiceURL":       "https://barbican.openstack.svc:9311",
-		"TransportURL":     string(transportURLSecret.Data["transport_url"]),
-		"LogFile":          fmt.Sprintf("%s%s.log", barbican.BarbicanLogPath, instance.Name),
-		"EnableSecureRBAC": instance.Spec.EnableSecureRBAC,
+		"ServiceURL": "https://barbican.openstack.svc:9311",
```

Comment:
which svc url is this? we create two services `barbican-internal` and `barbican-public`. there seems to be no `barbican`, also
* this should probably not be hard coded to the `openstack` namespace.  
* could it be non `https`? probably not recommended, but what happens when internal tls is disabled?

Reply:
investigating this ..

Reply:
Basically, this is the value that barbican puts in the href for a secret.  ie.  if you do an "openstack secret list", you see a value like "https://barbican.openstack.svc:9311/v1/secrets/sdfhhdfsk".  It turns out that doing "openstack secret get https://... " strips out the href part anyways, so it doesn't matter. 

The right thing to do though - and its what I've done in the latest version - is to set that parameter to "" (empty string).  This will cause barbican to populate the value from the HTTP request -so we'll get the right hostname and http/https.

Note - we do have to set it to "".  Not doing that will set the default value which is "http://localhost/..."